### PR TITLE
test: Update integration tests to use IAsyncLifetime

### PR DESCRIPTION
**Pull request by 96malhar**
*Description of changes:*
Update integration tests to use `IAsyncLifetime`. This allows the test initialization and disposal to happen asynchronously. 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: Libraries/test/TestServerlessApp.IntegrationTests/IntegrationTestContextFixture.cs [hunk 1]
```diff
@@ -14,24 +14,24 @@
...
-        public readonly string RestApiUrlPrefix;
-        public readonly string HttpApiUrlPrefix;
-        public readonly List<LambdaFunction> LambdaFunctions;
-        public readonly HttpClient HttpClient;
+        public LambdaHelper LambdaHelper;
+        public CloudWatchHelper CloudWatchHelper;
+        public string RestApiUrlPrefix;
+        public string HttpApiUrlPrefix;
+        public List<LambdaFunction> LambdaFunctions;
+        public HttpClient HttpClient;
```
Comment (ganeshnj) [inline Libraries/test/TestServerlessApp.IntegrationTests/IntegrationTestContextFixture.cs:29]: properties/fields like helpers which can be initialized in constructor should stay in constructor.

Move only the network call/async operations to `InitializeAsync`

**Review by ganeshnj — CHANGES_REQUESTED** [review]

File: Libraries/test/TestServerlessApp.IntegrationTests/IntegrationTestContextFixture.cs (hunk 1 above)
Comment (96malhar) [reply Libraries/test/TestServerlessApp.IntegrationTests/IntegrationTestContextFixture.cs]: Fixed

File: Libraries/test/TestServerlessApp.IntegrationTests/IntegrationTestContextFixture.cs [hunk 2]
```diff
@@ -14,61 +13,53 @@
...
+            LambdaHelper = new LambdaHelper(new AmazonLambdaClient());
+            CloudWatchHelper = new CloudWatchHelper(new AmazonCloudWatchLogsClient());
+            HttpClient = new HttpClient();
+        }
+
+        public async Task InitializeAsync()
         {
             var scriptPath = Path.Combine("..", "..", "..", "DeploymentScript.ps1");
-            CommandLineWrapper.Run($"pwsh {scriptPath}").GetAwaiter().GetResult();
+            await CommandLineWrapper.Run($"pwsh {scriptPath}");
```
Comment (ganeshnj) [inline Libraries/test/TestServerlessApp.IntegrationTests/IntegrationTestContextFixture.cs:44]: nit: `RunAsync`

**Review by ganeshnj — APPROVED** [review]

File: Libraries/test/TestServerlessApp.IntegrationTests/IntegrationTestContextFixture.cs (hunk 2 above)
Comment (96malhar) [reply Libraries/test/TestServerlessApp.IntegrationTests/IntegrationTestContextFixture.cs]: fixed

**Review by normj — APPROVED** [review]

**Pull request merged**